### DISCUSSION TITLE
idea: Collaborative Notes

### DISCUSSION THREAD

**Discussion by tgrrr** [codebuddies/frontend#141] [Ideas]
A new section of the CodeBuddies platform that allows everyone to capture and share their notes

#### What:

- Collaborative notes in realtime 
- Similar to Google Docs / Notion / Wiki
- Notes are grouped together. Eg. Python with Python, React with React, Docker with Docker, etc
- Automatically collect entries from Today I Learned (TIL) into a log, to be migrated (or at least searchable) in the collaborative docs

#### Why:

- Quicker and easier to collaborate with others
- Avoid repeating ourselves
- Tap the wealth of knowledge in CodeBuddies

#### My ideal version (yes, I'm checking my bias):

- Markdown
- Editable with common IDE's like VS Code, so we don't have to have an IDE and coding window open
- Makes publishing pages to a static blog a breeze
- Open sourced foundations like NextCloud

![](https://media.giphy.com/media/3o6Mbj2w67HnPQKgcE/giphy.gif)

I chatted about it with @bengineerdavis

and in this [CodeBuddies slack thread](https://codebuddies.slack.com/archives/C04BRN86J/p1598248035051300?thread_ts=1598248022.051200&cid=C04BRN86J)

**Comment by bengineerdavis** [comment]
The one danger of wikis is we would need enough active contributors to properly edit it from time to time and keep it DRY. It would be nice to integrate this collective knowledge into a PR-like process.

**Reply by tgrrr** [reply]
I dig git. 
I think we should simplify the process for beginners.

**Comment by lpatmo** [comment]
Do you think "notes for each resource" could be useful?

**Comment by bengineerdavis** [comment]
There's two main ways I would imagine we are using this:

One is to look up topics, and the other is combining topics when doing a
specific project. For instance, there's knowledge we want to know about
Django and React independently, and then there's extra stuff we have to
know when combining those two into a single application.

The Arch distro wiki could give us good ideas for how to structure it:
https://wiki.archlinux.org/
